relevanceai/http_client.py

```diff
@@ -100,3 +102,81 @@ def check_auth(self):
             return response.status_code == 200    
         except:
             raise Exception("Invalid auth details.")
+
+    def kmeans_cluster(self,
+        dataset_id: str,
+        vector_fields: list,
+        filters: List = [],
+        k: Union[None, int] = 10,
+        init: str = "k-means++",
+        n_init: int = 10,
+        max_iter: int = 300,
+        tol: float = 1e-4,
+        verbose: bool = True,
+        random_state = None,
+        copy_x: bool = True,
+        algorithm: str ="auto",
+        alias: str = "default",
+        cluster_field: str="_cluster_"
+    ):
+        """
+        This function performs all the steps required for Kmeans clustering:
+        1- Loads the data
+        2- Clusters the data
+        3- Updates the data with clustering info
+        4- Adds the centroid to the hidden centroid collection
+
+        input parameters:
+        dataset_id: name of the dataser
+        vector_fields: a list containing the vector field to be used for clustering
+        filters: a list to filter documents of the dataset,
+        k: K in Kmeans,
+        init: "k-means++" -> Kmeans algorithm parameter ,
+        n_init: number of reinitialization for the kmeans algorithm,
+        max_iter: max iteration in the kmeans algorithm,
+        tol: tol in the kmeans algorithm,,
+        verbose: bool = True,
+        random_state = None,
+        copy_x: bool = True,
+        algorithm: str ="auto",
+        alias: str = "default", string to be used in naming of the field showing the clustering results
+        cluster_field: str="_cluster_"
+
+        returns:
+        a list containing the calculated centroids
+
+        """
+        # load the documents
+        docs = self.get_all_documents(dataset_id=dataset_id, filters=filters, select_fields=vector_fields)
+
+        # Cluster
+        clusterer = KMeans(
+            k = k,
+            init = init,
+            n_init = n_init,
+            max_iter= max_iter,
+            tol= tol,
+            verbose = verbose,
+            random_state = random_state,
+            copy_x = copy_x,
+            algorithm = algorithm)
+        clustered_docs = clusterer.fit_documents(vector_fields, docs, return_only_clusters=True)
+
+        self.set_field_across_documents(
+            f"{cluster_field}.{vector_fields[0]}.{alias}",
+            [clustered_doc[cluster_field][vector_fields[0]][alias] for clustered_doc in clustered_docs],
+            docs
+        )
+
+        self.update_documents(dataset_id, docs)
+
+        # Update the centroid collection
+        centers = clusterer.get_centroid_docs()
+        self.services.cluster.centroids.insert(
+            dataset_id = dataset_id,
+            cluster_centers=centers,
+            vector_field=vector_fields[0],
+            alias= alias
+        )
```

Comment:
Again same situation here - what if `insert` errors? Based on what you are doing here - things will fail silently.